pulsar/producer_partition.go

```diff
@@ -1035,9 +1035,15 @@ func (p *partitionProducer) Send(ctx context.Context, msg *ProducerMessage) (Mes
 	}, true)
 
 	// wait for send request to finish
-	<-doneCh
-
-	return msgID, err
+	for {
```

Comment:
```suggestion
```
Hi, I don't think a loop is needed here.

Reply:
I also think this loop is useless.